test/integration/deploy_trigger_test.go

```diff
@@ -588,15 +589,19 @@ func TestTriggers_configChange(t *testing.T) {
 			t.Fatalf("expected watch event type %s, got %s", e, a)
 		}
 
-		if _, err := osClient.DeploymentConfigs(namespace).Update(config); err != nil {
-			return err
-		}
 		return nil
 	})
 	if retryErr != nil {
 		t.Fatal(retryErr)
 	}
 
+	if retryErr := kclient.RetryOnConflict(wait.Backoff{Steps: maxUpdateRetries}, func() error {
+		_, err := osClient.DeploymentConfigs(namespace).Update(newConfig)
```

Comment:
This is correct. There are also other places in our tests where we should retry on update conflicts: https://github.com/openshift/origin/pull/9995